{"ast":null,"code":"const space = process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID;\nconst accessToken = process.env.NEXT_PUBLIC_CONTENTFUL_ACCESS_TOKEN;\nconst environment = process.env.NEXT_PUBLIC_CONTENTFUL_ENVIRONMENT || 'master';\n\nasync function fetchContent(query) {\n  try {\n    const res = await fetch(`https://graphql.contentful.com/content/v1/spaces/${space}/environments/${environment}`, {\n      method: 'POST',\n      headers: {\n        'content-type': 'application/json',\n        authorization: `Bearer ${accessToken}`\n      },\n      // throw our query (a string) into the body directly\n      body: JSON.stringify({\n        query\n      })\n    });\n    const {\n      data\n    } = await res.json();\n    return data;\n  } catch (error) {\n    // add a descriptive error message first,\n    // so we know which GraphQL query caused the issue\n    console.error(`There was a problem retrieving entries with the query ${query}`);\n    console.error(error);\n  }\n}\n\nexport { fetchContent };","map":null,"metadata":{},"sourceType":"module"}